src/Microsoft.AspNet.Mvc.Core/Builder/RouteBuilderExtensions.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Routing;
+
+namespace Microsoft.AspNet.Builder
+{
+    /// <summary>
+    /// Extension methods for <see cref="IRouteBuilder"/>.
+    /// </summary>
+    public static class RouteBuilderExtensions
```

Comment:
Also suggest using these methods in the functional test sites or unit tests. Ignored in all tests at the moment.